plugins/outputs/kairosdb/kairosdb.go

```diff
@@ -0,0 +1,287 @@
+package kairosdb
+
+import (
+	"fmt"
+	"log"
+	"net"
+	"net/url"
+	"regexp"
+	"sort"
+	"strconv"
+	"strings"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/outputs"
+)
+
+var (
+	allowedChars = regexp.MustCompile(`[^a-zA-Z0-9-_./\p{L}]`)
+	hypenChars   = strings.NewReplacer(
+		"@", "-",
+		"*", "-",
+		`%`, "-",
+		"#", "-",
+		"$", "-")
+	defaultHttpPath  = "/api/v1/datapoints"
+	defaultSeperator = "_"
+)
+
+type KairosDB struct {
+	Prefix string
+
+	Host string
+	Port int
+
+	Username string
+	Password string
+
+	HttpBatchSize int // deprecated httpBatchSize form in 1.8
+	HttpPath      string
+
+	Debug bool
+
+	Separator string
+}
+
+var sampleConfig = `
+  ## prefix for metrics keys
+  prefix = "my.specific.prefix."
+
+  ## DNS name of the KairosDB server
+  ## Using "kairosdb.example.com" or "tcp://kairosdb.example.com" will use the
+  ## telnet API. "http://kairosdb.example.com" will use the Http API.
+  host = "http://kairosdb.example.com"
+
+  ## Port of the KairosDB server
+  port = 4242
+
+  ## HTTP basic authentication
+  ## Leave username or password empty to disable
+  username = ""
+  password = ""
+
+  ## Number of data points to send to KairosDB in Http requests.
+  ## Not used with telnet API.
+  http_batch_size = 50
+
+  ## URI Path for Http requests to KairosDB.
+  ## Used in cases where KairosDB is located behind a reverse proxy.
+  http_path = "/api/v1/datapoints"
+
+  ## Debug true - Prints KairosDB communication
+  debug = false
```

Comment:
Please don't add this! This is not useful for the average user and might confuse ppl wrt the telegraf command-line option.